### ISSUE TITLE
executed the test_local from github action 

### ISSUE THREAD

**Issue by tedteng** [gardener/gardenctl#447]
**Describe the bug**
I like to execute the test_local for e2e testing from GitHub action when any new commit push. the action which able to handle ssh testing in test_local and others eg `gardenctl` execution. and need feasible solution handle configuration file(garden dev kubeconfig)

**Comment by tedteng** [comment]
so far I am able to execute the action to check when PR changed. 
but 

![image](https://user-images.githubusercontent.com/42234376/99364700-44b6eb80-28f1-11eb-9f5c-6a736cc8679f.png)

**Comment by tedteng** [comment]
with the latest approach, will try to create kubectl image with oidc-login plugin which uses for GitHub action

```
Error: unknown command "oidc-login" for "kubectl"
```

**Comment by dansible** [comment]
We can't use the oidc kubeconfigs with the cicd as the 2-factor is meant for humans (not automation). We would need a normal kubeconfig with sufficient permissions to perform these tasks

**Comment by tedteng** [comment]
yes, you are right, 2-factor required humans input, and it seems to enable in Live only,  I never met in Dev required me to input the security code.  I thought I could found some silent mode since dev landscape never asking input 2-factor code. however, OIDC required an access browser. I can't handle it when in Github action.

now I am using the normal kubeconfig . as latest testing. It seems working well.
```
+ go run cmd/gardenctl/main.go target -g dev -p gardenctl -t gctl-aws
/usr/bin/xdg-open: 851: /usr/bin/xdg-open: www-browser: not found
/usr/bin/xdg-open: 851: /usr/bin/xdg-open: links2: not found
/usr/bin/xdg-open: 851: /usr/bin/xdg-open: elinks: not found
/usr/bin/xdg-open: 851: /usr/bin/xdg-open: links: not found
/usr/bin/xdg-open: 851: /usr/bin/xdg-open: lynx: not found
/usr/bin/xdg-open: 851: /usr/bin/xdg-open: w3m: not found
xdg-open: no method available for opening 'http://localhost:8000'
error: could not open the browser: exit status 3

Please visit the following URL in your browser manually: http://localhost:8000
```

![image](https://user-images.githubusercontent.com/42234376/100052908-e1313e80-2e59-11eb-809b-3db1565aa69e.png)

**Comment by tedteng** [comment]
manually check ssh AWS and Azure all green for  SSH testing,  but GCP failed in action (python v3.8 and Gcloud cli with the latest version).  will investigating.  
(skip openstack as is internal which not able to access from github.com)

![image](https://user-images.githubusercontent.com/42234376/100200382-79522500-2f39-11eb-9af6-74e7ed1060b5.png)

**Comment by tedteng** [comment]
after debug the current gcp ssh workaound in operate.go not support in github action yet. 

arguments
![image](https://user-images.githubusercontent.com/42234376/100301060-4e67df80-2fd2-11eb-8a3d-1be51dbc2529.png)

**Comment by tedteng** [comment]
achieved execute test_local through GitHub action. which also covers more testing cases mention in form  #393  (not include gcp as not support yet and ssh)however, it takes almost 10 mins. planning optimize it 
 
![image](https://user-images.githubusercontent.com/42234376/100691912-adf42f80-33c4-11eb-9e18-01c917ca44b2.png)

**Comment by tedteng** [comment]
as the latest changes, the total time cost around 3 mins after using container in Github action

![image](https://user-images.githubusercontent.com/42234376/100710711-15bd7100-33eb-11eb-8ada-0d6a2c6f8599.png)
